### ISSUE TITLE
Support new stash methods

### ISSUE THREAD

**Issue by ivoleitao** [ivoleitao/stash#22]
+ Map<K, V> getAll(Set<? extends K> keys);
 + void putAll(java.util.Map<? extends K, ? extends V> map);
 + void removeAll(Set<? extends K> keys);

**Comment by rlch** [comment]
@ivoleitao Any updates on this?

**Comment by ivoleitao** [comment]
Hello, still in the pipeline. I will try to add it on the next version

**Comment by naturallymitchell** [comment]
https://github.com/ivoleitao/stash/commit/cf6aac31f7da4923d4b6f5c507911a1cb57c82dd & https://github.com/ivoleitao/stash/commit/f700cfc050cf6caa56ff1917cea6b7b3804765a4 

@ivoleitao could you please push 4.0.2?

**Comment by ivoleitao** [comment]
I've found a concurrency problem, that I have to fix. It's an important one that was already present. In a nutshell the current solution had a race condition in the store creation / removal that appeared during the testing of getAll and putAll.

Unfortunately the solution is causing problems in sembast so I was unable to push it during the weekend.

I will try again next weekend.

**Comment by ivoleitao** [comment]
The problem is now solved in the latest version and the new features published. Unfortunately I had to do a breaking change to fix this concurrency issue. But It shouldn't create that much of a problem. Here’s what I did

* BREAKING CHANGE: The creation of vaults and caches now returns a `Future<Vault<T>>` and `Future<Cache<T>>`
* BREAKING CHANGE: The creation of stores now returns a `Future<XXXStore>` where XXX is the specific store

There was no other way to fix the issue without compromising the quality of the package
